src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -347,14 +368,29 @@ protected override void ProcessRecord()
                         WriteVerbose(reqVerboseMsg);
 
                         HttpResponseMessage response = GetResponse(client, request);
-                        response.EnsureSuccessStatusCode();
 
                         string contentType = ContentHelper.GetContentType(response);
                         string respVerboseMsg = string.Format(CultureInfo.CurrentCulture,
                             "received {0}-byte response of content type {1}",
                             response.Content.Headers.ContentLength,
                             contentType);
                         WriteVerbose(respVerboseMsg);
+
+                        if (!response.IsSuccessStatusCode)
+                        {
+                            string message = String.Format(CultureInfo.CurrentCulture, WebCmdletStrings.ResponseStatusCodeFailure,
+                                Convert.ToInt32(response.StatusCode).ToString(), response.ReasonPhrase);
```

Comment:
> Convert.ToInt32(response.StatusCode).ToString()

A minor comment: maybe just `(int)response.StatusCode` as in [HttpResponseMessage.cs](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs#L161)?
If we want to avoid boxing the int argument, then it should be `((int)response.StatusCode).ToString()`, but maybe it's an overkill 😄